### ISSUE TITLE
Add `bundle plugin list` command

### ISSUE THREAD

**Issue by kerrizor** [bundler/bundler#5467]
This command would list all the commands for installed plugins. A potential format for the output might be similar to `bundle plugin help`

```
$ bundle plugin list
Installed Plugins
  foo   # Find all gems whose name contains the word 'foo'
  bar   # Print the word 'bar' 16 times
  baz   # Make modem noises and draw an ANSI picture of a cactus
```

This would require CLI command, of course, as well as extending the plugin API to require that a "description" string be set (with a maximum length?) This could be enforced via `lib/bundler/plugin.rb#validate_plugin!`

**Comment by Shekharrajak** [comment]
I am interested in working on this issue. But I am not able to understand plugin system. I don't find documentation and examples. If there is any related links available let me know.

**Comment by kerrizor** [comment]
There are 2 kinds of plugins. One, the "old" way, is to simply have an executable in your $PATH that is named `bundler-<command>`. The "new" way is to use the Bundler::Plugin::API, which requires users to "register" the plugin by using the `bundle plugin install <command>` technique. An example of a working "new" style plugin can be found at http://github.com/kerrizor/bundler-dependency_graph

Bundler stores information about installed plugins in `~/.bundle/plugins` and I was thinking we could "easily" extend the Plugin::API DSL to allow plugins to declare their description (similar to how we declare the command here: https://github.com/kerrizor/bundler-dependency_graph/blob/master/lib/bundler/dependency_graph/graph_builder.rb#L4

**Comment by Shekharrajak** [comment]
Thanks @kerrizor .

**Comment by Shekharrajak** [comment]
Output must be list of commands (not the plugin name) ,right ?

**Comment by kerrizor** [comment]
My instinct says yes, a list of commands is more useful. 3 options:

- `bundle plugin list` shows plugins, `bundle plugin commands` shows the commands
- show a list of commands as sublists of plugins, something like:
```
$ bundle plugin list
Installed Plugins
  foo
  ---
    foo   # Find all gems whose name contains the word 'foo'

  bar
  ---
    bar   # Print the word 'bar' 16 times
    baz   # Make modem noises and draw an ANSI picture of a cactus
```

- list plugins "inline" so more like this
```
$ bundle plugin list
Installed Plugins
  foo   (foo)   # Find all gems whose name contains the word 'foo'
  bar   (bar)   # Print the word 'bar' 16 times
  baz   (bar)   # Make modem noises and draw an ANSI picture of a cactus
```

What do you think?

**Comment by kerrizor** [comment]
BTW, whatever we pick, it is super low stakes, since a change will have little impact on users. :)

**Comment by andrew** [comment]
It looks like there's a plugin list command on master: https://github.com/bundler/bundler/blob/master/lib/bundler/plugin.rb#L52-L68 which came from https://github.com/bundler/bundler/pull/6120 but it doesn't appear in any of the released versions

**Comment by deivid-rodriguez** [comment]
Closed by https://github.com/bundler/bundler/pull/6120. Will be released with bundler 2.1.0.
